compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -229,7 +227,10 @@ impl<'p, 'tcx: 'p> RustcMatchCheckCtxt<'p, 'tcx> {
                     } else {
                         let variant =
                             &adt.variant(RustcMatchCheckCtxt::variant_index_for_adt(&ctor, *adt));
-                        let tys = cx.list_variant_nonhidden_fields(ty, variant).map(|(_, ty)| ty);
+                        let tys = cx
+                            .list_variant_nonhidden_fields(ty, variant)
```

Comment:
Are there any calls to list_variant_nonhidden_fields that *don't* filter out the skipped fields?